src/compressed_tensors/transform/factory/hadamard.py

```diff
@@ -105,6 +106,11 @@ def forward(self, value: Tensor) -> Tensor:
             weight = weight.T
 
         return (
-            apply_transform_weight(weight, value, self.args.location, self.module_type)
+            apply_transform_weight(
+                weight.to(self.scheme.precision),
+                value.to(self.scheme.precision),
+                self.args.location,
+                self.module_type,
+            )
             / self._scale
-        )
+        ).to(value.dtype)
```

Comment:
Just to see if I understand, we do the forward pass with one precision but then return / maintain the original precision?